spring-cloud-netflix-eureka-client/src/test/java/org/springframework/cloud/netflix/eureka/serviceregistry/EurekaServiceRegistryTests.java

```diff
@@ -130,4 +139,59 @@ void eurekaClientGetStatusNoInstance() {
 		assertThat(map).hasSize(1).containsEntry("status", UNKNOWN.toString());
 	}
 
+	@Test
```

Comment:
I was able to reproduce the issue and fixed it by executing the register-call inside another Thread that waits for a short while. However I can see that the usage of Thread.sleep has the potential to lead to flaky tests.

The test and subject code could also be reworked to check if a (newly created and package-protected) method is called in the subject code if **asyncClientInitialization** is true. This would eliminate any flakyness from the test but diminish its potential to detect issues with synchronicity.